src/main/java/com/ciandt/techgallery/service/impl/CronServiceImpl.java

```diff
@@ -239,4 +240,73 @@ public void sendEmailToEndorseds() {
     cronJobsDao.add(cronJob);
   }
 
+  /**
+   * Method that groups endorsements by technologies.
+   * 
+   * @param endorsementsList list of endorsements.
+   * @return map with grouped endorsements.
+   */
+  private Map<String, List<Endorsement>> groupEndorsementsByTechnology(
+      List<Endorsement> endorsementsList) {
+    Map<String, List<Endorsement>> groupedEndorsements = new HashMap<String, List<Endorsement>>();
+    for (Endorsement endorsement : endorsementsList) {
+      String key = endorsement.getTechnologyEntity().getName();
+
+      List<Endorsement> group = groupedEndorsements.get(key);
+      if (group == null) {
+        group = new ArrayList<Endorsement>();
+      }
+      group.add(endorsement);
+      // add or update the Map
+      groupedEndorsements.put(key, group);
+    }
+    return groupedEndorsements;
+  }
+
+  /**
+   * Method that set the user activities by grouping endorsements by technology.
+   * 
+   * @param endorsementsList list of user endorsements.
+   * @param activities activities for the email.
+   */
+  private void setUserActivities(List<Endorsement> endorsementsList,
+      TechGalleryActivitiesEmailTemplateTO activities) {
+    Map<String, List<Endorsement>> groupedEndorsements =
+        groupEndorsementsByTechnology(endorsementsList);
+    _LOG.info("Technology hashmap size: " + groupedEndorsements.size());
+
+    for (Map.Entry<String, List<Endorsement>> entry : groupedEndorsements.entrySet()) {
+      List<Endorsement> technologyEndorsements = entry.getValue();
+      if (technologyEndorsements != null && technologyEndorsements.size() > 0) {
+        TechnologyActivitiesEmailTemplateTO endorsementActivity;
+        // single endorsement
+        if (technologyEndorsements.size() == 1) {
+          _LOG.info("Only one endorsement for Technology: " + entry.getKey());
+          Endorsement endorsement = technologyEndorsements.get(0);
```

Comment:
@Surerus Please extract this block before the 'if'. This is also used on the 'else' block